app/js/spatial/geo.js

```diff
@@ -95,6 +97,56 @@ export function watchLocation(callback) {
   );
 }
 
+/*
+  I think we should probably have this bit of mapping enumerateTilesAround() be executed only once 
+  and passed to different functions, because as it is now, this first part is done twice for both this function
+  and the announcer function to get all the tiles features.
+  - Kris
```

Comment:
Agreed; we should probably have a separate tile service so we don't need to copy-and-paste code. But at least as currently written the tile-fetching code does anticipate being called by multiple uncoordinated asynchronous threads, so this doesn't result in unnecessary network traffic or database duplication.